_posts/2022-07-24-opensearch-plugin-zips-now-in-maven-repo.markdown

```diff
@@ -0,0 +1,95 @@
+
+## Motivation 
+
+Before OpenSearch 2.1, plugin zips used as dependencies could not be downloaded dynamically during runtime because plugin zips were not a part of the version-controlled Maven system. The only mechanism for plugin downloads was each plugins' respective Java jars through Maven coordinates. This system forced users who wanted more control over their OpenSearch plugin configuration to use a dependency plugin zip built on a developer desktop instead of a more reliable version-controlled plugin. 
+
+Furthermore, to facilitate the plugin availability as a dependency, tests executed against as part the OpenSearch build process from zip were not accurate, as each local build zip had to find the plugin repo in order to ensure that plugin's availability. These restrictions were challenging to our community since using plugin zips as separate isolated components via a cached mechanism proved to be impossible. 
+
+## Benefits of Maven
+
+With Maven, plugin zips can now be retrieved by:
+
+- Downloading each plugin directly using their respective Maven coordinates .
+- Using clickstream from the central Maven repo, which can be cached later to a local Maven repo.
+- Fetching the development `SNAPSHOT` version with same Maven `groupID` as [org.opensearch.plugin](https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/plugin/).
+
+Using OpenSearch plugin zips through Maven offers the following benefits:
+
+- Plugins zip in the central Maven repo are already signed with `.asc`, `.md5`, `.sha1`, `.sha256`, and `.sha512` extensions.
+- Users are no longer required to to check in zips to any `src/` files since zips can be fetched with the right `groupID`, `artifactID`, and `version`.
+- Tests and CI workflows can directly run against zips from the Maven repo instead of requiring a manual download. 
+
+## Maven zip publication with Gradle
+
+OpenSearch publishes plugin zips using a custom Gradle plugin, `opensearch.pluginzip`. With OpenSearch 2.1, all OpenSearch Gradle-supported plugins create a new task, `publishPluginZipPublicationToZipStagingRepository`. The task performs all the heavy lifting for users by:
+
+- Identifying the distribution plugin zip.
+- Setting the Maven coordinates.
+- Generating the POM file.
+- Updating with the user-generated POM fields.
+- Publishing the zip to your Maven repo.
+
+You can find more details about the inner workings of OpenSearch plugins in the [opensearch-plugins repo](https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#opensearchpluginzip).
+
+**Figure 1**: Workflow that ships generated plugin zips to Maven
+
+![Figure 1: Workflow that ship zips to maven]({{ site.baseurl }}/assets/media/blog-images/2022-07-24-opensearch-plugin-zips-now-in-maven-repo/figure1.png){: .img-fluid }**Figure 1**: Workflow that ships generated plugin zips to Maven
+
+## Consume plugin in zips
+
+You can fetch plugin in zips in three different ways:
```

Comment:
Should "plugin" be "plugins" in both instances, or should there be an article in front of "plugin" in both instances? Change the colon to a period.